### DISCUSSION TITLE
how to use optional parameters?

### DISCUSSION THREAD

**Discussion by ommpawin** [leafsphp/leaf#165] [Q&A] [answered]
Can I use Optional Parameters Routing like this

```
Route::get('/user/{name?}', function ($name = null) {
    return $name;
});
 
Route::get('/user/{name?}', function ($name = 'John') {
    return $name;
});
```

Ref: https://laravel.com/docs/9.x/routing#parameters-optional-parameters

**Comment by mychidarko** [comment, marked as answer]
@ommpawin This is also possible with Leaf. You can find the documentation @ https://leafphp.dev/docs/routing/sub-patterns.html
